src/nvme/types.h

```diff
@@ -9007,4 +9025,324 @@ struct nvme_ns_mgmt_host_sw_specified {
+ * @NVME_LM_SEQIND_FIRST:		This command is the first of a sequence of two or more
+ *					Migration Send commands
+ * @NVME_LM_SEQIND_LAST:		This command is the last command of a sequence of two or
+ *					more Migration Send commands
+ * @NVME_LM_SEQIND_ENTIRE:		This Migration Send command is the only command and contains
+ *					the entire controller state for this management operation
+ * @NVME_LM_SEQIND_SHIFT:		Shift amount to set SEQIND
+ * @NVME_LM_SEQIND_MASK:		Mask to set SEQIND
+ * @NVME_LM_MIGRATION_SEND_CSUUIDI_SHIFT: Shift amount to set Controller State UUID Index (CSUUIDI)
+ * @NVME_LM_MIGRATION_SEND_CSUUIDI_MASK:  Mask to set CSUUIDI
+ * @NVME_LM_MIGRATION_SEND_CSVI_SHIFT:	Shift amount to set Controller State Version Index (CSVI)
+ * @NVME_LM_MIGRATION_SEND_CSVI_MASK:	Mask to set CSVI
+ * @NVME_LM_MIGRATION_RECV_CSUIDXP_SHIFT: Shift amount to set Controller State UUID Index Parameter
+ *					  (CSUIDXP)
+ * @NVME_LM_MIGRATION_RECV_CSUIDXP_MASK:  Mask to set CSUIDXP
+ * @NVME_LM_MIGRATION_RECV_CSUUIDI_SHIFT: Shift amount to set Controller State UUID Index (CSUUIDI).-
+ * @NVME_LM_MIGRATION_RECV_CSUUIDI_MASK:  Mask to set CSUUIDI
+ * @NVME_LM_MIGRATION_RECV_CSVI_SHIFT:	Shift amount to set Controller State Version Index (CSVI)
+ * @NVME_LM_MIGRATION_RECV_CSVI_MASK:	Mask to set CSVI
+ * @NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED: Controller Suspended
+ * @NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED_SHIFT: Shift to get Controller Suspended value in
+ *						 completion dword0 of Migration Receive command
+ * @NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED_MASK:  Mask to get Controller Suspended value in
+ *						 completion dword0 of Migration Receive command
+ */
+enum nvme_lm_migration {
+	NVME_LM_MIGRATION_CNTLID_SHIFT		= 0,
+	NVME_LM_MIGRATION_CNTLID_MASK		= 0xffff,
+	NVME_LM_MIGRATION_UIDX_SHIFT		= 0,
+	NVME_LM_MIGRATION_UIDX_MASK		= 0x7f,
+
+	/* Migration Send - Suspend */
+	NVME_LM_DUDMQ				= 1,
+	NVME_LM_DUDMQ_SHIFT			= 31,
+	NVME_LM_DUDMQ_MASK			= 0x1,
+	NVME_LM_STYPE_NOTIFICATION		= 0,
+	NVME_LM_STYPE_SUSPEND			= 1,
+	NVME_LM_STYPE_SHIFT			= 16,
+	NVME_LM_STYPE_MASK			= 0xff,
+
+	/* Migration Send - Set Controller State */
+	NVME_LM_SEQIND_NOT_FIRST_NOT_LAST	= 0,
+	NVME_LM_SEQIND_FIRST			= 1,
+	NVME_LM_SEQIND_LAST			= 2,
+	NVME_LM_SEQIND_ENTIRE			= 3,
+	NVME_LM_SEQIND_SHIFT			= 16,
+	NVME_LM_SEQIND_MASK			= 0xf,
+	NVME_LM_MIGRATION_SEND_CSUUIDI_SHIFT	= 24,
+	NVME_LM_MIGRATION_SEND_CSUUIDI_MASK	= 0xff,
+	NVME_LM_MIGRATION_SEND_CSVI_SHIFT	= 16,
+	NVME_LM_MIGRATION_SEND_CSVI_MASK	= 0xff,
+
+	/* Migration Receive - Get Controller State */
+	NVME_LM_MIGRATION_RECV_CSUIDXP_SHIFT	= 24,
+	NVME_LM_MIGRATION_RECV_CSUIDXP_MASK	= 0xff,
+	NVME_LM_MIGRATION_RECV_CSUUIDI_SHIFT	= 16,
+	NVME_LM_MIGRATION_RECV_CSUUIDI_MASK	= 0xff,
+	NVME_LM_MIGRATION_RECV_CSVI_SHIFT	= 16,
+	NVME_LM_MIGRATION_RECV_CSVI_MASK	= 0xff,
+	NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED	= 1,
+	NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED_SHIFT = 0,
+	NVME_LM_MIGRATION_RECV_CTRL_SUSPENDED_MASK = 0x1,
+};
```

Comment:
same comment as above, this seems to mix the definition from different commands into one enum.